hardware_interface/test/test_resource_manager.cpp

```diff
@@ -1569,7 +1679,20 @@ TEST_F(ResourceManagerTestReadWriteError, handle_error_on_hardware_write)
   // check write methods failures
   check_read_or_write_failure(
     std::bind(&TestableResourceManager::write, rm, _1, _2),
-    std::bind(&TestableResourceManager::read, rm, _1, _2), WRITE_FAIL_VALUE);
+    std::bind(&TestableResourceManager::read, rm, _1, _2),
+    hardware_interface::test_constants::WRITE_FAIL_VALUE);
+}
+
+TEST_F(ResourceManagerTestReadWriteError, handle_deactivate_on_hardware_write)
+{
+  setup_resource_manager_and_do_initial_checks();
+
+  using namespace std::placeholders;
+  // check write methods failures
+  check_read_or_write_deactivate(
+    std::bind(&TestableResourceManager::write, rm, _1, _2),
+    std::bind(&TestableResourceManager::read, rm, _1, _2),
+    hardware_interface::test_constants::WRITE_DEACTIVATE_VALUE);
```

Comment:
Aren't you here missing testing failures on `read`?

```suggestion
  check_read_or_write_deactivate(
    std::bind(&TestableResourceManager::write, rm, _1, _2),
    std::bind(&TestableResourceManager::read, rm, _1, _2),
    hardware_interface::test_constants::WRITE_DEACTIVATE_VALUE);
  check_read_or_write_deactivate(
    std::bind(&TestableResourceManager::read, rm, _1, _2),
    std::bind(&TestableResourceManager::write, rm, _1, _2),
    hardware_interface::test_constants::READ_DEACTIVATE_VALUE);
```

Reply:
fixed in 9d7df03